tests/unit/test_routes.py

```diff
@@ -9,54 +9,232 @@
+
+    for resource in response.json['data']:
+        assert (resource.get('category') == "Back End Dev")
+
+    # TODO: Filter by updated_after
+    # (Need to figure out how to manually set last_updated and created_at)
+
+
+def test_languages(module_client, module_db):
+    client = module_client
 
-    response = create_resource(client)
+    response = client.get('api/v1/languages')
+    for language in response.json['data']:
+        assert (isinstance(language.get('id'), int))
+        assert (isinstance(language.get('name'), str))
+        assert (len(language.get('name')) > 0)
+
+
+def test_categories(module_client, module_db):
+    client = module_client
+
+    response = client.get('api/v1/categories')
+
+    for category in response.json['data']:
+        assert (isinstance(category.get('id'), int))
+        assert (isinstance(category.get('name'), str))
+        assert (len(category.get('name')) > 0)
+
+
+def test_update_votes(module_client, module_db):
+    client = module_client
+    vote_direction = 'upvote'
+    id = 1
+
+    # Check voting on a valid resource
+    data = client.get(f"api/v1/resources/{id}").json['data']
+    response = client.put(f"/api/v1/resources/{id}/{vote_direction}", follow_redirects=True)
+    initial_votes = data.get(f"{vote_direction}s")
+
+    assert (response.status_code == 200)
+    assert (response.json['data'].get(f"{vote_direction}s") == initial_votes + 1)
+
+    # Check voting on an invalid resource
+    id = 'waffles'
+    response = client.put(f"/api/v1/resources/{id}/{vote_direction}", follow_redirects=True)
+    assert (response.status_code == 404)
+    assert (response.json.get('errors')[0].get('code') == "not-found")
+
+    # Check voting on a resource that doesn't exist
+    too_high = 99999999
+    response = client.put(f"/api/v1/resources/{too_high}/{vote_direction}", follow_redirects=True)
+    assert (response.status_code == 404)
+    assert (response.json.get('errors')[0].get('code') == "not-found")
+
+
+##########################################
+## Authenticated Routes
+##########################################
+
+
+def test_get_api_key(module_client, module_db, patched_request):
+    client = module_client
+
+    response = client.post('api/v1/apikey', json = dict(
+        email="test@example.org",
+        password="supersecurepassword"
+    ))
+
+    assert (response.status_code == 200)
+    assert (response.json['data'].get('email') == "test@example.org")
+    assert (isinstance(response.json['data'].get('apikey'), str))
+
+
+def test_create_resource(module_client, module_db, patched_request):
+    client = module_client
+
+    apikey = get_api_key(client)
+    response = create_resource(client, apikey)
     assert (response.status_code == 200)
     assert (isinstance(response.json['data'].get('id'), int))
     assert (response.json['data'].get('name') == "Some Name")
 
 
-def test_update_resource(session_app, function_session, session_db):
```

Comment:
These three functions should be marked with a comment that they aren't currently in use but we'd like to go back to using them when we can avoid _always_ using the persistent DB